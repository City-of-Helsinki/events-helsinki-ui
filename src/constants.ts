export const CATEGORIES = {
  CULTURE: "culture",
  DANCE: "dance",
  FOOD: "food",
  INFLUENCE: "influence",
  MISC: "misc",
  MOVIE: "movie",
  MUSEUM: "museum",
  MUSIC: "music",
  NATURE: "nature",
  SPORT: "sport",
  THEATRE: "theatre"
};

export const DISTRICTS = {
  ALPPIHARJU: "kaupunginosa:alppiharju",
  EIRA: "kaupunginosa:eira",
  ETU_TÖÖLÖ: "kaupunginosa:etu-t%C3%B6%C3%B6l%C3%B6",
  HAAGA: "kaupunginosa:haaga",
  HERMANNI: "kaupunginosa:hermanni",
  HERTTONIEMI: "kaupunginosa:herttoniemi",
  KAARELA: "kaupunginosa:kaarela",
  KAARTINKAUPUNKI: "kaupunginosa:kaartinkaupunki",
  KAIVOPUISTO: "kaupunginosa:kaivopuisto",
  KALLIO: "kaupunginosa:kallio",
  KAMPPI: "kaupunginosa:kamppi",
  KATAJANOKKA: "kaupunginosa:katajanokka",
  KLUUVI: "kaupunginosa:kluuvi",
  KONALA: "kaupunginosa:konala",
  KOSKELA: "kaupunginosa:koskela",
  KRUUNUNHAKA: "kaupunginosa:kruununhaka",
  KULOSAARI: "kaupunginosa:mustikkamaa-korkeasaari,kaupunginosa:kulosaari",
  KUMPULA: "kaupunginosa:kumpula",
  KÄPYLÄ: "kaupunginosa:k%C3%A4pyl%C3%A4",
  LAAJASALO:
    "kaupunginosa:laajasalo,kaupunginosa:villinki,kaupunginosa:santahamina",
  LAAKSO: "kaupunginosa:laakso",
  LAUTTASAARI: "kaupunginosa:lauttasaari",
  LÄNSISATAMA: "kaupunginosa:l%C3%A4nsisatama",
  MALMI: "kaupunginosa:malmi",
  MEILAHTI: "kaupunginosa:meilahti",
  MELLUNKYLÄ: "kaupunginosa:mellunkyl%C3%A4",
  MUNKKINIEMI: "kaupunginosa:munkkiniemi",
  OULUNKYLÄ: "kaupunginosa:oulunkyl%C3%A4",
  PAKILA: "kaupunginosa:pakila",
  PITÄJÄMÄKI: "kaupunginosa:pit%C3%A4j%C3%A4nm%C3%A4ki",
  PUKINMÄKI: "kaupunginosa:pukinm%C3%A4ki",
  PUNAVUORI: "kaupunginosa:punavuori",
  RUSKEASUO: "kaupunginosa:ruskeasuo",
  SUOMENLINNA: "kaupunginosa:suomenlinna,kaupunginosa:ulkosaaret",
  SUURMETSÄ: "kaupunginosa:suurmets%C3%A4",
  SUUTARILA: "kaupunginosa:suutarila",
  SÖRNÄINEN: "kaupunginosa:s%C3%B6rn%C3%A4inen",
  TAKA_TÖÖLÖ: "kaupunginosa:taka-t%C3%B6%C3%B6l%C3%B6",
  TAMMISALO: "kaupunginosa:tammisalo",
  TAPANINKYLÄ: "kaupunginosa:tapaninkyl%C3%A4",
  ULLANLINNA: "kaupunginosa:ullanlinna",
  VALLILA: "kaupunginosa:vallila",
  VANHAKAUPUNKI: "kaupunginosa:vanhakaupunki",
  VARTIOKYLÄ: "kaupunginosa:vartiokyl%C3%A4",
  VARTIOSAARI: "kaupunginosa:vartiosaari",
  VIIKKI: "kaupunginosa:viikki",
  VUOSAARI: "kaupunginosa:vuosaari",
  ÖSTERSUNDOM:
    "kaupunginosa:%C3%B6stersundom,kaupunginosa:salmenkallio,kaupunginosa:talosaari,kaupunginosa:karhusaari,kaupunginosa:ultuna"
};

export const DATE_TYPES = {
  ALL: "all",
  THIS_WEEK: "this_week",
  TODAY: "today",
  TOMORROW: "tomorrow",
  WEEKEND: "weekend"
};

export const KEYWORD_TYPES = {
  DISTRICT: "district",
  KEYWORD: "keyword",
  PLACE: "place",
  YSO: "yso"
};

export enum SUPPORT_LANGUAGES {
  FI = "fi",
  SV = "sv",
  EN = "en"
}

export const TARGET_GROUPS = {
  CHILDREN: "children",
  YOUNG_PEOPLE: "young_people"
};
