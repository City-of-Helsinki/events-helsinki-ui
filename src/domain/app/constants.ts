export enum ROUTES {
  ABOUT = '/about',
  COLLECTION = '/collection/:id',
  COLLECTIONS = '/collections',
  EVENTS = '/events',
  EVENT = '/event/:id',
  HOME = '/home',
  HOME_PREVIEW = '/home/:id',
}
