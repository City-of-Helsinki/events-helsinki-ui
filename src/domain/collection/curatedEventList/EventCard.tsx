import React from "react";

import LoadingSpinner from "../../../common/components/spinner/LoadingSpinner";
import { useEventDetailsQuery } from "../../../generated/graphql";
import SimilarEventCard from "../../event/SimilarEventCard";
import styles from "./eventCard.module.scss";

interface Props {
  eventId: string;
}

const EventCard: React.FC<Props> = ({ eventId }) => {
  const { data: eventData, loading } = useEventDetailsQuery({
    variables: {
      id: eventId
    }
  });

  return (
    <div className={styles.eventCard}>
      <LoadingSpinner isLoading={loading}>
        {eventData && <SimilarEventCard event={eventData.eventDetails} />}
      </LoadingSpinner>
    </div>
  );
};

export default EventCard;
