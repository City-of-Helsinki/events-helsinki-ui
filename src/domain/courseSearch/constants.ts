export const DEFAULT_SEARCH_FILTERS = {
  categories: [],
  dateTypes: [],
  divisions: [],
  end: null,
  isFree: false,
  onlyOngoingCourses: false,
  places: [],
  start: null,
  text: [],
};
