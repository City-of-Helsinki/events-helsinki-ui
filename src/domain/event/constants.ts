export const mockEventData = {
  eventDetails: {
    description: {
      en: "description en",
      fi: "description fi",
      sv: "description sv"
    },
    endTime: null,
    externalLinks: [
      {
        link: "https://www.facebook.com/events/12345687/",
        name: "extlink_facebook"
      }
    ],
    id: "123",
    images: [
      {
        id: "60578",
        name: "piha-2.jpg",
        url: "https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg"
      }
    ],
    inLanguage: [
      {
        id: "fi",
        name: {
          en: "Finnish",
          fi: "suomi",
          sv: "finska"
        },
        translationAvailable: true
      }
    ],
    infoUrl: {
      en:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      fi:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      sv:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080"
    },
    keywords: [],
    location: {
      addressLocality: {
        en: "locality en",
        fi: "locality fi",
        sv: "locality sv"
      },
      divisions: [],
      email: "location@info.com",
      infoUrl: {
        en: "http://www.infourl.com",
        fi: "http://www.infourl.com",
        sv: "http://www.infourl.com"
      },
      name: {
        en: "name en",
        fi: "name fi",
        sv: "name sv"
      },
      position: {
        coordinates: [25.01497, 60.250507],
        type: "Point"
      },
      postalCode: "01234",
      streetAddress: {
        en: "streetAddress en",
        fi: "streetAddress fi",
        sv: "streetAddress sv"
      },
      telephone: {
        en: "044 123 4567",
        fi: "044 123 4567",
        sv: "044 123 4567"
      }
    },
    name: {
      en: "name en",
      fi: "name fi",
      sv: "name sv"
    },
    offers: [
      {
        description: null,
        infoUrl: null,
        isFree: true,
        price: null
      }
    ],
    provider: null,
    publisher: "provider:123",
    shortDescription: {
      en: "short description en",
      fi: "short description fi",
      sv: "short description sv"
    },
    startTime: "2019-12-13T08:00:00Z"
  }
};

// Amount of the similar events
export const SIMILAR_EVENTS_AMOUNT = 8;

export const EVENT_KEYWORD_BLACK_LIST = [
  "helmet:10592", // Yleinen
  "helmet:10689", // Helsinki
  "helmet:11916", // Infonäytöt (Helsinki)
  "kulke:53", // Suomi
  "kulke:54", // Ruotsi
  "kulke:55", // Englanti
  "kulke:616", // Näkyy TV-sovelluksessa
  "kulke:671", // Normaali
  "kulke:673" // Suomi (TV)
];
