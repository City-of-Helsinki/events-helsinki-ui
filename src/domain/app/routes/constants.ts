import { MAPPED_PLACES } from '../../eventSearch/constants';

// Remember also update static urls to updateSitemap
export type ROUTE_TYPES = {
  ABOUT: '/about';
  ACCESSIBILITY: '/accessibility';
  COLLECTION: '/collection/:slug';
  COLLECTIONS: '/collections';
  EVENTS: '/events';
  EVENT: '/event/:id';
  COURSES: '/courses';
  COURSE: '/course/:id';
  HOME: '/home';
  HOME_PREVIEW: '/home/:id';
  EVENT_PLACE: string;
};

export const ROUTES: ROUTE_TYPES = {
  ABOUT: '/about',
  ACCESSIBILITY: '/accessibility',
  COLLECTION: '/collection/:slug',
  COLLECTIONS: '/collections',
  EVENTS: '/events',
  EVENT: '/event/:id',
  COURSES: '/courses',
  COURSE: '/course/:id',
  HOME: '/home',
  HOME_PREVIEW: '/home/:id',
  EVENT_PLACE: `/:place(${Object.keys(MAPPED_PLACES).join('|')})`,
};
