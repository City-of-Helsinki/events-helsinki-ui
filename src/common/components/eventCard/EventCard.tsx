import { IconArrowRight } from "hds-react";
import React from "react";
import { useTranslation } from "react-i18next";
import { useLocation } from "react-router";
import { Link } from "react-router-dom";

import EventKeywords from "../../../domain/event/eventKeywords/EventKeywords";
import LocationText from "../../../domain/event/eventLocation/EventLocationText";
import {
  getEventImageUrl,
  getEventPrice
} from "../../../domain/event/EventUtils";
import { EventInList } from "../../../domain/event/types";
import useLocale from "../../../hooks/useLocale";
import getDateRangeStr from "../../../util/getDateRangeStr";
import getLocalisedString from "../../../util/getLocalisedString";
import getTimeRangeStr from "../../../util/getTimeRangeStr";
import IconLink from "../link/IconLink";
import SrOnly from "../srOnly/SrOnly";
import styles from "./eventCard.module.scss";

interface Props {
  event: EventInList;
}

const SimilarEventCard: React.FC<Props> = ({ event }) => {
  const { search } = useLocation();
  const { t } = useTranslation();
  const locale = useLocale();

  const imageUrl = getEventImageUrl(event);
  const name = event.name;
  const startTime = event.startTime;
  const endTime = event.endTime;

  const eventUrl = React.useMemo(() => {
    return `/${locale}/event/${event.id}${search}`;
  }, [event.id, locale, search]);

  return (
    <div className={styles.eventCard}>
      <Link
        aria-hidden={true}
        aria-label={t("commons.eventCard.ariaLabelLink", {
          name: getLocalisedString(name, locale)
        })}
        className={styles.imageWrapper}
        style={{ backgroundImage: `url(${imageUrl})` }}
        tabIndex={-1}
        to={eventUrl}
      >
        <div className={styles.tagWrapperDesktop}>
          <div className={styles.priceWrapper}></div>

          <div className={styles.categoryWrapper}>
            <EventKeywords
              event={event}
              showIsFree={true}
              showKeywords={false}
            />
          </div>
        </div>
      </Link>
      <div className={styles.infoWrapper}>
        <div>
          <div className={styles.categoryWrapperMobile}>
            <EventKeywords
              event={event}
              showIsFree={true}
              showKeywords={false}
            />
          </div>
          <div className={styles.textWrapper}>
            <div className={styles.eventDateAndTime}>
              {!!startTime &&
                `${getDateRangeStr(
                  startTime,
                  endTime,
                  locale,
                  false
                )} ${getTimeRangeStr(startTime, endTime, locale)}`}
            </div>
            <Link
              aria-hidden="true"
              className={styles.eventName}
              tabIndex={-1}
              to={eventUrl}
            >
              {getLocalisedString(name, locale)}
            </Link>
            {/* Event name for screen readers  */}
            <SrOnly>{getLocalisedString(name, locale)}</SrOnly>
            <div className={styles.eventLocation}>
              <LocationText
                event={event}
                showDistrict={false}
                showLocationName={true}
              />
            </div>
            <div className={styles.eventPrice}>
              {getEventPrice(
                event,
                locale,
                t("eventSearch.event.offers.isFree")
              )}
            </div>
          </div>
        </div>
        <div className={styles.buttonWrapper}>
          <IconLink
            aria-label={t("commons.eventCard.ariaLabelLink", {
              name: getLocalisedString(name, locale)
            })}
            icon={<IconArrowRight />}
            to={eventUrl}
          />
        </div>
      </div>
    </div>
  );
};

export default SimilarEventCard;
