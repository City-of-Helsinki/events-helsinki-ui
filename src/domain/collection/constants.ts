import { mockEventData } from "../event/constants";

export const mockCollection = {
  collectionDetails: {
    curatedEvents: [
      `http://localhost:3000/fi/event/${mockEventData.eventDetails.id}`
    ],
    curatedEventsTitle: {
      en: "At least visit these",
      fi: "Käy ainakin näissä",
      sv: "Besök åtminstone dessa"
    },
    description: {
      en:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      fi:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      sv:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus."
    },
    eventListQuery:
      "http://localhost:3000/en/events?categories=music&districts=kaupunginosa%3Aetu-t%C3%B6%C3%B6l%C3%B6",
    eventListTitle: {
      en: "All the best events of the fall",
      fi: "Kaikki syksyn parhaat tapahtumat",
      sv: "Höstens bästa händelser"
    },
    id: "1",
    linkText: {
      en: "Read more on the project website",
      fi: "Lue lisää hankkeen omilta sivuilta",
      sv: "Läs mer på projektets webbplats"
    },
    linkUrl: {
      en: "http://www.google.com",
      fi: "http://www.google.com",
      sv: "http://www.google.com"
    },
    similarCollectionsTitle: {
      en: "Check also these",
      fi: "Katso myös nämä",
      sv: "Se dessa också"
    },
    socialMediaDescription: {
      en: "We put together the best foliage for the fall",
      fi: "Kokosimme parhaat tärpit syksylle",
      sv: "Vi sätter ihop det bästa bladverket för hösten"
    },
    subtitles: {
      en: "Subtitles en",
      fi: "Subtitles fi",
      sv: "Subtitles sv"
    },
    title: {
      en: "The most bustling tricks of the fall",
      fi: "Syksyn riehakkaimmat riennot",
      sv: "Höstens mest livliga knep"
    }
  }
};

// Amount of the similar collections
export const SIMILAR_COLLECTIONS_AMOUNT = 8;
