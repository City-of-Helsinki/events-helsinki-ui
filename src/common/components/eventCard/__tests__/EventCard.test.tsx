import { shallow } from "enzyme";
import React from "react";
import routeData, { MemoryRouter } from "react-router";

import { mockEventData } from "../../../../domain/event/constants";
import EventCard from "../EventCard";

const mockLocation = {
  hash: "",
  pathname: "/fi/home",
  search: "",
  state: ""
};
beforeEach(() => {
  jest.spyOn(routeData, "useLocation").mockReturnValue(mockLocation);
});

it("EventCard matches snapshot", () => {
  const container = shallow(
    <MemoryRouter>
      <EventCard event={mockEventData.eventDetails} />
    </MemoryRouter>
  );
  expect(container.html()).toMatchSnapshot();
});
