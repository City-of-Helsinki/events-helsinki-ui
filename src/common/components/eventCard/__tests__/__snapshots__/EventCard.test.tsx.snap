// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard matches snapshot 1`] = `"<div class=\\"eventCard\\"><a aria-hidden=\\"true\\" aria-label=\\"Siirry tapahtumaan: name fi\\" class=\\"imageWrapper\\" style=\\"background-image:url(https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg)\\" tabindex=\\"-1\\" href=\\"/fi/event/123\\"><div class=\\"tagWrapperDesktop\\"><div class=\\"priceWrapper\\"></div><div class=\\"categoryWrapper\\"></div></div></a><div class=\\"infoWrapper\\"><div><div class=\\"categoryWrapperMobile\\"></div><div class=\\"textWrapper\\"><div class=\\"eventDateAndTime\\"> 13. joulukuuta 2019 10.00</div><a aria-hidden=\\"true\\" class=\\"eventName\\" tabindex=\\"-1\\" href=\\"/fi/event/123\\">name fi</a><div class=\\"srOnly\\">name fi</div><div class=\\"eventLocation\\">name fi, streetAddress fi, locality fi</div><div class=\\"eventPrice\\">Ilmainen</div></div></div><div class=\\"buttonWrapper\\"><a aria-label=\\"Siirry tapahtumaan: name fi\\" class=\\"iconLink default\\" href=\\"/fi/event/123\\"><div class=\\"iconWrapper\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\"></polygon></g></svg></div></a></div></div></div>"`;
